services/github/github-directory-file-count.service.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const path = require('path')
+const Joi = require('joi')
+const { InvalidParameter } = require('..')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.alternatives(
+  Joi.object({}).required(),
+  Joi.array()
+    .items(
+      Joi.object({
+        path: Joi.string().required(),
+        type: Joi.string().required(),
+      })
+    )
+    .required()
+)
+
+const queryParamSchema = Joi.object({
+  type: Joi.any().valid('dir', 'file'),
+  extension: Joi.string(),
+})
+
+module.exports = class GithubDirectoryFileCount extends ConditionalGithubAuthV3Service {
+  static category = 'size'
+
+  static route = {
+    base: 'github/directory-file-count',
+    pattern: ':user/:repo/:path*',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'GitHub repo directory file count',
+      pattern: ':user/:repo/:path',
+      namedParams: { user: 'badges', repo: 'shields', path: 'services' },
+      staticPreview: this.render({ count: 10 }),
+      documentation,
+    },
+    {
+      title: 'GitHub repo directory file count (dir type)',
```

Comment:
```suggestion
      title: 'GitHub repo directory count',
```

Will probably also want a non-required `:path` param example for this one and the files as well

Reply:
I've added the examples.